### PR TITLE
correct logic in spg unit deliver

### PR DESCRIPTION
**Pull request by JacobHarrisonONS**

### CODE REVIEW

**Review by samreidons — APPROVED** [review]
this is great

File: src/main/java/uk/gov/ons/census/fwmt/jobservice/service/routing/spg/SpgCreateUnitDeliverRouter.java [hunk 1]
```diff
@@ -22,7 +22,8 @@ public Boolean isValid(FwmtActionInstruction ffu, GatewayCache cache) {
     try {
       // relies on the validation of: SpgRouter, SpgCreateRouter
       return ffu.getAddressLevel().equals("U")
-          && ffu.isHandDeliver();
+          && ffu.isHandDeliver()
+          || (cache.existsInFwmt && cache.delivered);
```
Comment (sdiaz1000) [inline src/main/java/uk/gov/ons/census/fwmt/jobservice/service/routing/spg/SpgCreateUnitDeliverRouter.java:26]: at this point could cache be null and get a npe? what do the other rules do

Comment (sdiaz1000) [reply src/main/java/uk/gov/ons/census/fwmt/jobservice/service/routing/spg/SpgCreateUnitDeliverRouter.java:26]: is it checked somewhere else?

Comment (sdiaz1000) [reply src/main/java/uk/gov/ons/census/fwmt/jobservice/service/routing/spg/SpgCreateUnitDeliverRouter.java:26]: oh the catch catches it

**Review by sdiaz1000 — APPROVED** [review]

**Pull request merged**